default.py

```diff
@@ -930,6 +971,70 @@ def download_subtitles(url):
     return outfile
 
 
+def SignInBBCiD():
+    #Below is required to get around an ssl issue
+    requests.packages.urllib3.disable_warnings()
+    sign_in_url="https://ssl.bbc.co.uk/id/signin"
+    
+    username=ADDON.getSetting('bbc_id_username')
+    password=ADDON.getSetting('bbc_id_password')
+    
+    post_data={
+               'unique': username, 
+               'password': password, 
+               'rememberme':'0'}
+    r = OpenURLPost(sign_in_url, post_data)
+    if (r.status_code == 302):
+        xbmcgui.Dialog().notification("BBCiD Sign In", "Successful")
+    else:
+        xbmcgui.Dialog().notification("BBCiD Sign In", "Failed - check settings")
+
+
+def SignOutBBCiD():
+    sign_out_url="https://ssl.bbc.co.uk/id/signout"
+    OpenURL(sign_out_url)
+
+
+def StatusBBCiD():
+    status_url="https://ssl.bbc.co.uk/id/status"
+    html=OpenURL(status_url)
+    if("You are signed in." in html):
+        xbmcgui.Dialog().ok("BBC iD", "You are signed in.")
+    else:
+        xbmcgui.Dialog().ok("BBC iD", "You are currently not signed in.")
+
+
+def ListWatching():
+    #StatusBBCiD()
+    #xbmcgui.Dialog().notification("BBC iPlayer", "List Watching not implemented", xbmcgui.NOTIFICATION_ERROR)
+    identity_cookie = None
+    for cookie in GetCookies():
+        if (cookie.name == 'IDENTITY'):
+            identity_cookie = cookie.value
+            break
+    url = "https://ibl.api.bbci.co.uk/ibl/v1/user/watching?identity_cookie=%s" % identity_cookie
+    html = OpenURL(url)
+    json_data = json.loads(html)
+    watching_list = json_data.get('watching').get('elements')
+    for watching in watching_list:
+        programme = watching.get('programme')
+        name = programme.get('title')
+        programme_id = programme.get('id')
+        AddMenuEntry(name, programme_id, 121, '', '', '')
```

Comment:
I am not sure mode 121 for all entries in Favourites and Watching is correct. Mode 121 should only be used to create a directory with multiple entries. If there are single entries, which can be played directly, CheckAutoplay should be used.